src/genotyper.cc

```diff
@@ -108,15 +108,47 @@ class FormatFieldHelper : public IFormatFieldHelper {
         return ans;
     }
 
+    Status get_default_value(T* val) {
+        if (field_info.default_type == DefaultValueFiller::ZERO) {
+            *val = 0;
+        } else if (field_info.default_type == DefaultValueFiller::MISSING) {
+            switch (field_info.type) {
+                case RetainedFieldType::INT:
+                    *val = bcf_int32_missing;
+                    break;
+                case RetainedFieldType::FLOAT:
+                {
+                    float tmp = 0;
```

Comment:
Even though it's an unreachable code path at runtime, I am just a little bit nervous by the syntactic assignment of an int variable to a float value which is lossy...try the union variable htslib uses (it's a trick to make the compiler assume the same bytes of memory are a different data type depending on how the variable is used), I think that should be pretty clean if it works out. https://github.com/samtools/htslib/blob/b6aa0e6e2350ca7836c275082f1d680fdca9e335/htslib/vcf.h#L827